protostar/cheatable_starknet/callable_hint_locals/callable_hint_local.py

```diff
@@ -38,11 +39,15 @@ def _build(self) -> Callable:
         ...
 
     def build(self):
-        def wrapper(*args: Any, **kwargs: Any):
+        def wrapper(*args: Any, **kwargs: Any) -> ExecutionResult:
             try:
                 result = self._build()(*args, **kwargs)
                 return ValidExecution(ok=result)
             except TransactionRevertException as ex:
-                return InvalidExecution(err_code=encode_shortstring(ex.message[:31]))
+                return InvalidExecution(
+                    err_code=encode_shortstring(
+                        hashlib.md5(ex.message.encode()).hexdigest()[:31]
```

Comment:
the point of calculating a hash here is that we may have different error messages with the same prefix of 31+ signs, like: `an error occurred here and there ... (enough signs to reach the 31 length threshold) ..., the cause is: [different cause for multiple errors]`